pkg/auth/basic_service.go

```diff
@@ -153,17 +158,39 @@ func (s *BasicAuthService) CreateCredentials(ctx context.Context, username strin
 }
 
 func (s *BasicAuthService) AddCredentials(ctx context.Context, username, accessKeyID, secretAccessKey string) (*model.Credential, error) {
-	currCreds, err := s.listUserCredentials(ctx, username)
+	_, err := s.GetUser(ctx, username)
+	if err != nil {
+		return nil, err
+	}
+
+	currCreds, err := s.listUserCredentials(ctx, SuperAdminKey, BasicPartitionKey, "")
 	if err != nil {
 		return nil, err
 	}
 	// TODO (niro): Support swap?
 	if len(currCreds) >= MaxCredentialsPerUser {
 		return nil, fmt.Errorf("exceeded number of allowed credentials: %w", ErrInvalidRequest)
 	}
+
+	// Handle user import flow from previous auth service
+	if accessKeyID != "" && secretAccessKey == "" {
+		return s.importUserCredentials(ctx, username, accessKeyID)
+	}
+
 	return s.addCredentials(ctx, username, accessKeyID, secretAccessKey)
 }
 
+func (s *BasicAuthService) importUserCredentials(ctx context.Context, username, accessKeyID string) (*model.Credential, error) {
+	creds, err := s.listUserCredentials(ctx, username, model.PartitionKey, accessKeyID)
+	if err != nil {
+		return nil, err
+	}
+	if len(creds) < 1 {
+		return nil, fmt.Errorf("no credentials found for user (%s): %w", username, ErrNotFound)
+	}
+	return s.addCredentials(ctx, username, creds[0].AccessKeyID, creds[0].SecretAccessKey)
```

Comment:
IIUC - if the user has more than two credentials, it will return not found! please check this and add a test if that's the case

Reply:
We are using listUserCredentials with the accessKeyID as a prefix, this will return a list of no more than 1 items. Either the key exists or it doesn't

Reply:
I missed the prefix, thanks!